### DISCUSSION TITLE
Release Profiling

### DISCUSSION THREAD

**Discussion by alazifk** [godotengine/godot-proposals#5479] [General Discussions]
If my understanding is correct we normally profile GDScript in debug mode and you cannot use the Profiler for release builds. This seems problematic. The naïve assumption is that release mode is just a faster version of debug mode, but the reality is that one implementation can be slower than another in debug mode and faster in release mode. Which means you have to do profiling in an exported release. Which I don't think a lot of people do, I think a lot of the time the implementation that runs faster when you hit play survives.

Is this correct? Should tools for release profiling be provided and release profiling be made easier? I imagine something like a "Run in release mode" Button and a "Runtime Profiler" node.

**Comment by Calinou** [comment]
For release profiling, I think something like https://github.com/godotengine/godot-proposals/issues/5346 will give better results anyway. The built-in profiler will stay as it's easier to set up, but it's optimized for usability, not the most extensive results.
